### PR TITLE
Preview: Add a diff endpoint

### PR DESCRIPTION
**Pull request by nik9000**
I've found that it can be kind of annoying trying to find out which
files changed when looking up a docs preview. This adds a page at
`/diff` that contains links to all of the modified pages.

I used chunked transfer and stream parsing the result from git because
that felt like the right thing to do here but it did make the parsing a
bit more complex.

I also chose to use the `-z` flag when calling git because it emits a
format that isn't jumbled when files move. It has many ascii-nul
terminated records which means I couldn't really use any off the shelf
nodejs parsing libraries.

### CODE REVIEW

**Comment by nik9000** [comment]
The page is totally unstyled but does the job:
2019-07-19-094822_1463x640_scrot

**Comment by nik9000** [comment]
If you want to make your laptop lift off do
```
preview/test.sh
<wait for it to come online>
curl staging.localhost:8000/diff
```

That makes `git` take a few seconds to get the results to you:
```
22054 root      20   0 1137816 346088 119020 R  99.7   2.1   0:34.74 git diff-tree -z --find-renames --numstat staging --                                    
```

**Comment by nik9000** [comment]
I'll also want to bump the docker image version in this PR. I'll push an update in a few minutes.

**Comment by nik9000** [comment]
> I'll also want to bump the docker image version in this PR. I'll push an update in a few minutes.

Done.

@haywood, could you have a look at this when you get a chance? I think it'll make the previews easier to use.

File: preview/preview.js [hunk 1]
```diff
@@ -54,51 +60,197 @@ const requestHandler = (request, response) => {
...
+
+  const pipeline = child.stdout.pipe(handle);
+  rig_handlers(child, response, pipeline, response => {
+    response.write(handleChunk());
+    if (chunk !== '') {
+      console.error('unprocessed results from git', chunk);
+      response.write(`  <li>Unprocessed results from git: <pre>${chunk}</pre>`);
+      response.status = 500;
+    } else if (!completeSuccess) {
+      response.write(`  <li>Error processing some entries from git. See logs.`);
```
Comment (haywoood) [inline preview/preview.js]: I might just be missing where these `li` tags are being closed.

**Comment by nik9000** [comment]
Thanks for the review @haywoood!

**Pull request merged**